server/chat-plugins/youtube.ts

```diff
@@ -129,6 +143,27 @@ export class YoutubeInterface {
 		if (!(id in this.data.channels)) return this.getChannelData(id, username);
 		return Promise.resolve({...this.data.channels[id]});
 	}
+	async getVideoData(link: string): Promise<VideoData | null> {
+		const id = this.getId(link);
+		const raw = await Net(`${ROOT}videos`).get({
+			query: {part: 'snippet,statistics', id, key: Config.youtubeKey},
+		});
+		const res = JSON.parse(raw);
+		if (!res || !res.items || res.items.length < 1) return null;
+		const video = res.items[0];
+		return {
+			title: video.snippet.title,
+			id,
```

Comment:
```suggestion
			id as ID,
```

Reply:
This isnt actually an ID, it's a video ID (which has hyphens etc in it)

Reply:
So hardcasting isnt good

Reply:
im confused why it was labeled that in the interface though

Reply:
It's an easy mistake to make; it was labeled "ID" after all.

Reply:
ya i wasnt faulting anyone, just saying